HexBox.WinUI/HexBox.cs

```diff
@@ -1560,26 +1601,49 @@ protected override void OnPointerMoved(PointerRoutedEventArgs e)
         {
             base.OnPointerMoved(e);
 
-            switch (_HighlightState)
-            {
-            case SelectionArea.Data:
-            case SelectionArea.Text:
+            if (_HighlightState != SelectionArea.None)
             {
                 var position = e.GetCurrentPoint(_Canvas).Position;
-
                 var currentMouseOverOffset = ConvertPositionToOffset(position);
 
-                if (currentMouseOverOffset >= SelectionStart)
+                switch (_HighlightState)
+                {
+                    case SelectionArea.Data:
+                    case SelectionArea.Text:
+                    {
+                        if (currentMouseOverOffset >= SelectionStart)
+                        {
+                            SelectionEnd = currentMouseOverOffset + _BytesPerColumn;
+                        }
+                        else
+                        {
+                            SelectionEnd = currentMouseOverOffset;
+                        }
+                        break;
+                    }
+                    case SelectionArea.Address:
+                    {
+                        if (currentMouseOverOffset >= SelectionStart)
```

Comment:
Idea: If selection switches from up to down and other way: we need to move the start point if `SelectStart == CurrentRow`.